src/common.ts

```diff
@@ -652,8 +653,10 @@ export function addBlockName(
       scope.scopeError = recordScopeError(parser, Errors.DuplicateBinding, name);
     } else if (
       context & Context.OptionsWebCompat &&
-      value & BindingKind.FunctionLexical &&
-      origin & Origin.BlockStatement
+      (context & Context.Strict) === 0 &&
+      origin & Origin.BlockStatement &&
+      value === BindingKind.FunctionLexical &&
+      kind === BindingKind.FunctionLexical
```

Comment:
> It is a Syntax Error if the [LexicallyDeclaredNames](https://262.ecma-international.org/#sec-static-semantics-lexicallydeclarednames) of [StatementList](https://262.ecma-international.org/#prod-StatementList) contains any duplicate entries, **unless the [source text matched by](https://262.ecma-international.org/#sec-algorithm-conventions-syntax-directed-operations) this production is not [strict mode code](https://262.ecma-international.org/#sec-strict-mode-code) and the duplicate entries are only bound by FunctionDeclarations.**

webcompat mode, not strict mode, in block, both duplicate entries are FunctionDeclarations

Reply:
`===` makes sure they are not AsyncFunctionLexical.